common/json.hpp

```diff
@@ -18,23 +18,85 @@
 #include <common/error.hpp>
 #include <common/expected.hpp>
 
-namespace json {
+#include "config.h"
+
+#ifdef MENDER_COMMON_USE_NLOHMANN_JSON
+#include <nlohmann/json.hpp>
+#endif
+
+using namespace std;
+
+namespace mender::common::json {
 
 enum JsonErrorCode {
 	NoError = 0,
-	KeyNoExist,
+	ParseError,
+	KeyError,
+	IndexError,
 	TypeError,
 };
 using JsonError = mender::common::error::Error<JsonErrorCode>;
 
 class Json {
 public:
 	using ExpectedJson = mender::common::expected::Expected<Json, JsonError>;
-	void hello_world();
+	using ExpectedString = mender::common::expected::Expected<string, JsonError>;
+	using ExpectedInt = mender::common::expected::Expected<int, JsonError>;
+	using ExpectedBool = mender::common::expected::Expected<bool, JsonError>;
+	using ExpectedSize = mender::common::expected::Expected<size_t, JsonError>;
+
+	Json() = default;
+
+	string Dump(const int indent = 2) const;
+
+	ExpectedJson Get(const char *child_key) const;
+	ExpectedJson operator[](const char *child_key) const {
+		return this->Get(child_key);
+	}
+	ExpectedJson Get(string child_key) const {
+		return this->Get(child_key.data());
+	}
+	ExpectedJson operator[](string child_key) const {
+		return this->Get(child_key.data());
```

Comment:
I *think* you can use `const string &` arguments to these functions, can't you? Unless `data` requires non-const, but I don't think so as long as it's returning `const char *`.

Reply:
yup!